app/forms/waste_carriers_engine/tier_check_form.rb

```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class TierCheckForm < BaseForm
     attr_accessor :temp_tier_check
 
+    validates :temp_tier_check, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.temp_tier_check = @transient_registration.temp_tier_check
+      self.temp_tier_check = transient_registration.temp_tier_check
```

Comment:
NIT Adding a line between super and this as per other changes

```suggestion
      
      self.temp_tier_check = transient_registration.temp_tier_check
```